packages/block-editor/src/components/block-switcher/index.js

```diff
@@ -137,7 +228,7 @@ export const BlockSwitcherDropdownMenu = ( { clientIds, blocks } ) => {
 		!! possibleBlockTransformations.length && canRemove && ! isTemplate;
 	const hasPossibleBlockVariationTransformations =
 		!! blockVariationTransformations?.length;
-	const hasPatternTransformation = !! patterns?.length && canRemove;
+
 	if (
 		! hasBlockStyles &&
 		! hasPossibleBlockTransformations &&
```

Comment:
Unfortunately we can't do this refactoring because we could end up with en empty Dropdown. This highlights an existing bug though because we should add the `&& ! hasPatternTransformation` check here. If we have no styles or block transformations, but we have patterns, now we do not render the Dropdown.

Reply:
But it exists in trunk :D

Reply:
Ok, I guess we need lighter `has*` selectors?

Reply:
Yeah probably some `has` selectors. I actually introduced a private one [yesterday](https://github.com/WordPress/gutenberg/pull/57006/files#diff-23e851c3077ed95d23bb1669da8bafbae3f1e2e952897987e45b49555004dd83R254) - not for this case. We would need something that filters first by `blockType` and then parses, but what concerns me a bit is that with [lazy loading ](https://github.com/WordPress/gutenberg/pull/57006#discussion_r1424977196)the patterns parsing will have to be refactored anyway. So maybe wait a bit for these? 🤔

Reply:
>But it exists in trunk :D

I [opened a PR](https://github.com/WordPress/gutenberg/pull/57047) for the bug in trunk